merkledag/merkledag.go

```diff
@@ -460,18 +458,19 @@ func EnumerateChildrenAsync(ctx context.Context, ds DAGService, c *cid.Cid, visi
 			if inProgress == 0 && next == nil {
 				return nil
 			}
-		case nc := <-out:
-			if nc.Err != nil {
-				return nc.Err
-			}
-
-			if next == nil {
-				next = nc.Node
-				send = feed
-			} else {
-				todobuffer = append(todobuffer, nc.Node)
+		case cids := <-out:
```

Comment:
if out was changed to be a channel of nodes, this could look like:
```go
case nd := <-out:
  for _, lnk := range nd.Links() {
    if next == nil {
      next = lnk.Cid
      send = feed
      continue
    }
    todobuffer = append(todobuffer, lnk.Cid)
  }
```